lib/finitomata.ex

```diff
@@ -442,6 +443,7 @@ defmodule Finitomata do
           end
         else
           err ->
+            state = %State{state | error: err}
```

Comment:
After another fresh look, I hesitate to say what would be better, to keep a stale error forever (current,) to clean it up right before line 420 or to pass it to `safe_on_failure/3` _only_.

Thoughts?

This should be clarified before we merge the code.

Reply:
I've commented below on "cleaning up the error" part.
With regards to passing it only to `safe_on_failure/3` - maybe even better would be to pass it to `safe_on_failure/4`? As in - add new attribute to `safe_on_failure` which is the `err`? So we can keep state cleaner (and even have an easier way to pattern match the error).
What do you think?

Reply:
Also, if we pass only to `on_failure` (be it `/3` or `/4`) - the replies for soft events (line 448) and `:continue` for ensure entry (line 455) will lack the error.